### ISSUE TITLE
Automatically close appeals in "awaiting reply" after N days

### ISSUE THREAD

**Issue by MarcoAurelioWM** [UTRS2/utrs#605]
Feature request: If a ticket is in "AWAITING_REPLY" status for more than N days (to be defined), automatically close the appeal as "expired" (or create a new status e.g. "closed_noreply") and notify the user about the fact so he can create a new appeal.

**Comment by dqwiki** [comment]
Legit and easy to do. Might do it in this weekend's sprint of code changes.

**Comment by MarcoAurelioWM** [comment]
@dqwiki If possible I think a new status "no reply" or similar would be preferable as "declined" has a very specific meaning, as well as "expired". Appeals closed due to lack of reply should be catalogued as such IMHO. Thanks.

**Comment by dqwiki** [comment]
> @dqwiki If possible I think a new status "no reply" or similar would be preferable as "declined" has a very specific meaning, as well as "expired". Appeals closed due to lack of reply should be catalogued as such IMHO. Thanks.

I see very little benefit gained from this, and it would require going through many many places in the code and tweaking database queries to account for the new status - aka a fair amount of work and bugs introduced if I missed one. What is the extra benefit that you might be suggesting with this?

**Comment by MarcoAurelioWM** [comment]
> I see very little benefit gained from this, and it would require going through many many places in the code and tweaking database queries to account for the new status - aka a fair amount of work and bugs introduced if I missed one. What is the extra benefit that you might be suggesting with this?

I was mostly thinking on the ability to filter these just in case, but if the cost of adding a new status outweights the benefits of it then I certainly withdraw the proposal. I still belive however that an auto close as expired after N days without appelant follow-up for appeals in "awaiting reply" functionality would be helpful, which I think you're not declining?

Thank you.

**Comment by dqwiki** [comment]
Correct, I still plan to do work on the main task here, and they will be marked as expired when doing so.

**Comment by DeirgeDel** [comment]
As someone affected by this situation I'd like to add a comment about I feel you should be implementing this flow.
* Being blocked is mental stress point.
* Consider cost/risk/benefit to appellants of continuing with UTRS versus giving up on the process and returning as a sock an new user(s) or anon. ip.
:* This cost/risk/benefit will vary between individuals but may stress people more with more content added and if vandalism/disruption to that content is not being addressed.
* Filling out an appeal is a mental stress point.
* Having appeals open for a long time is stressful.  I've held a public library book open for over 15 months to resume editing an article. I've also similarly moaned about the length of time deletion discussions have been ongoing on VFD.
* If administrator's have failed to close a UTRS within a certain time period then talk page access at least should be automatically restored.  There is even a case for the block to be automatically rescinded if that repeatedly is repeatedly restored.
* Every appeal closed really needs a time before an fresh appeal can be entered.  If that time is unreasonably long a sock will likely be created. And there may even be admins who so want that so they detect socks! Unlikely ... but possible.
* In most cases a block should NEVER be longer than 6 months to allow the standard offer.  Unless the appellant simply goes into a rant.  A one or two week cool down is sometimes appropriate.  If no specific reason is or a specific refactor suggestion that is also perhaps a two week block.  Too long a block and there's a high chance of creating a sock.  However setting clear boundaries for naughty children is often more effective.

I've begun to drift but the key point is to try to get the workflow so respect is shown to the appellant so they are kept updated with a reasonable reply within a certain period. if they reasonably trying to get back to normal content editing.  In my use case its known I have a grievance against my blocking administrator (Starting with the that they and their friends were having in-jokes about a gun video in an ANI thread prior to blocking me and thus are in my perspective involved and also for denying me access to question a user later identified as a sockpuppet which may well have exposed them on the spot). I'm not an angel and COMPETANCE IS REQUIRED continues to stress my brain.  Thankyou.
